storage/range.go

```diff
@@ -1391,22 +1394,26 @@ func (r *Range) InternalPushTxn(batch engine.Engine, args *proto.InternalPushTxn
 	// pusherWins bool is true in the event the pusher prevails.
 	var pusherWins bool
 
-	// If there's no incoming transaction, the pusher is
-	// non-transactional. We make a random priority, biased by
-	// specified args.Header().UserPriority in this case.
+	// If there's no incoming transaction, the pusher is non-transactional.
+	// We make a random priority, biased by specified
+	// args.Header().UserPriority in this case.
 	var priority int32
 	if args.Txn != nil {
 		priority = args.Txn.Priority
 	} else {
-		priority = proto.MakePriority(args.GetUserPriority())
+		// Make sure we have a deterministic random number when generating
+		// a priority for this txn-less request, so all replicas see same priority.
+		rNG := rand.New(rand.NewSource(int64(reply.PusheeTxn.Priority)))
```

Comment:
Yeah, that's why I didn't use rng...rNG looked slightly better. Can't use "r", "rng", "rand"...sigh.